serverlessworkflow/modules/ROOT/pages/security/authention-support-for-openapi-services.adoc

```diff
@@ -1 +1,382 @@
+quarkus.oidc-client.oauth_example.client-id=kogito-app
+quarkus.oidc-client.oauth_example.grant.type=client
+quarkus.oidc-client.oauth_example.credentials.client-secret.method=basic
+quarkus.oidc-client.oauth_example.credentials.client-secret.value=secret
+----
+
+The configuration suffix `quarkus.oidc-client.oauth_example` is exclusive for the schema defined in the specification file and the scheme name is sanitized by applying the rules described <<sanitize_note, above>>.
+
+For this to work you **must** add the https://quarkus.io/guides/security-openid-connect-client#oidc-client-filter[Quarkus OIDC Client Filter Extension] to your project:
+
+[source, xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-oidc-client-filter</artifactId>
+</dependency>
+----
+
+[TIP]
+====
+You can add the extension using the Quarkus CLI:
+[source, bash]
+$ quarkus extension add quarkus-oidc-client-filter
+====
+
+== Authorization Token Propagation
+
+The authorization token propagation can be used with OpenAPI operations that are secured with a security scheme of type `oauth2` or `http bearer`.
+When configured, you can propagate the authorization tokens passed to your Serverless Workflow at the time it was created, along the OpenAPI invocations executed by that service.
+These propagations must be configured individually for each of the corresponding security schemes, in that way, you can do a fine-tuned configuration of which invocations require the propagation.
+Similarly to the other security scheme configurations, the token propagation apply to all the operations secured with the same scheme.
+
+Following the <<_oauth2_authentication>> example presented above, these are the supported configurations:
+
+[frame=all]
+|===
+|Property Key |Example
+
+| `quarkus.openapi-generator.[filename].auth.[security_scheme_name].token-propagation=[true,false]`
+| `quarkus.openapi-generator.security_example_json.auth.oauth_example.token-propagation=true` +
+ +
+Enables the token propagation for all the operations that are secured with the `oauth-example` scheme in the `security-example.json` file. By default, the standard HTTP Authorization header is propagated.
+
+| `quarkus.openapi-generator.[filename].auth.[security_scheme_name].header-name=[http_header_name]`
+| `quarkus.openapi-generator.security_example_json.auth.oauth_example.header-name=MyHeaderName` +
+ +
+States that the authorization token to propagate will be read from the HTTP header `MyHeaderName` instead of the standard HTTP `Authorization` header.
+
+|===
+
+[IMPORTANT]
+====
+The authorization tokens are propagated as long the Serverless Workflow doesn't reach a "waiting" state, for example to wait for an event to arrive. When the Serverless Workflow is resumed the tokens will be no longer propagated.
+====
+
+== What's next?
+
+This guide covered the authentication types that are supported to access the OpenAPI service operations and the respective configurations.
```

Comment:
It can probably be removed.

Reply:
I mean, this: "This guide covered the authentication types that are supported to access the OpenAPI service operations and the respective configurations."

can be removed.

Reply:
ah, yes, now I understand. Yeah, it's again a styling criteria, many times when you write an article, blog, paper, etc, you finish with a short "summary", "phrase", etc, and then, you write the conclusions, next steps, etc.